src/ts/component/page/main/date.tsx

```diff
@@ -0,0 +1,278 @@
+					rootId={object.chatId} 
+				/>
+
+				{content}
+
+				<Footer component="mainObject" {...this.props} />
+			</div>
+		);
+	};
+
+	onCalendar = () => {
+		const rootId = this.getRootId();
+		const object = S.Detail.get(rootId, rootId, ['timestamp']);
+
+		S.Menu.open('dataviewCalendar', {
+			element: '#calendar-icon',
+			horizontal: I.MenuDirection.Center,
+			data: {
+				value: object.timestamp,
+				canEdit: true,
+				canClear: false,
+				onChange: (value: number) => {
+					C.ObjectDateByTimestamp(U.Router.getRouteSpaceId(), value, (message: any) => {
+						U.Object.openAuto(message.details);
+					});
+				},
+			},
+		});
+	};
+
+	componentDidMount () {
+		this._isMounted = true;
+		this.open();
+	};
+
+	componentDidUpdate () {
+		this.open();
+		this.checkDeleted();
+	};
+
+	componentWillUnmount () {
+		this._isMounted = false;
+		this.close();
+	};
+
+	checkDeleted () {
+		const { isDeleted } = this.state;
+		if (isDeleted) {
+			return;
+		};
+
+		const rootId = this.getRootId();
+		const object = S.Detail.get(rootId, rootId, []);
+
+		if (object.isDeleted) {
+			this.setState({ isDeleted: true });
+		};
+	};
+
+	open () {
+		const rootId = this.getRootId();
+
+		if (this.id == rootId) {
+			return;
+		};
+
+		this.close();
+		this.id = rootId;
+		this.setState({ isDeleted: false, isLoading: true });
+
+		C.ObjectOpen(rootId, '', U.Router.getRouteSpaceId(), (message: any) => {
+			if (!U.Common.checkErrorOnOpen(rootId, message.error.code, this)) {
+				return;
+			};
+
+			const object = S.Detail.get(rootId, rootId, []);
+			if (object.isDeleted) {
+				this.setState({ isDeleted: true, isLoading: false });
+				return;
+			};
+
+			this.refHeader?.forceUpdate();
+			this.refHead?.forceUpdate();
+
+			this.setState({ isLoading: false });
+			this.loadCategory();
+		});
+	};
+
+	close () {
+		if (!this.id) {
+			return;
+		};
+
+		const { isPopup, match } = this.props;
+		
+		let close = true;
+		if (isPopup && (match.params.id == this.id)) {
+			close = false;
+		};
+		if (close) {
+			Action.pageClose(this.id, true);
+		};
+	};
+
+	loadCategory() {
+		const { space, config } = S.Common;
+		const rootId = this.getRootId();
+
+		const object = S.Detail.get(rootId, rootId);
+
+		C.RelationListWithValue(space, object.id, (message: any) => {
+			this.setState({ relations: message.relations });
+		});
+	}
+
+	prevDate = () => {
+		const rootId = this.getRootId();		
+		const object = S.Detail.get(rootId, rootId, ['timestamp']);
+		C.ObjectDateByTimestamp(U.Router.getRouteSpaceId(), object.timestamp - 24 * 60 * 60, (message: any) => {
+			U.Object.openAuto(message.details);
+		});
+	};
+
+	nextDate = () => {
+		const rootId = this.getRootId();		
+		const object = S.Detail.get(rootId, rootId, ['timestamp']);
+		console.log(object.timestamp);
+		C.ObjectDateByTimestamp(U.Router.getRouteSpaceId(), object.timestamp + 24 * 60 * 60, (message: any) => {
+			U.Object.openAuto(message.details);
+		});
+	};
```

Comment:
prev/nextDate logic can be combined into one method, space id should be defined as `const { space } = S.Common` from store.